Docker.md

```diff
@@ -0,0 +1,264 @@
+# Docker for .NET
+
+## Docker overview
+
+Motivation behind using docker is having a reproducible environment for application development and deployment.
+
+Instead of installing libraries, tools, configuration, etc. - everywhere the application runs - a [docker image](#docker-image) which contains everything application needs to run (tools, add configuration files/resources, libraries, etc.). This image is then consistent between development machine, cloud host, VPS, etc. [Docker container](#docker-container) is self-contained and isolated - other applications or different versions of some tool running on host won't affect container execution. Likewise once something is added to an image, even if the remote source is gone, the image contains all the files it needs to run.
```

Comment:
The sentence structure is too complex here. Let's rephrase:

FROM
Instead of setting up the application dependencies in each environment (tools, libraries, configuration) we can use a [docker image](#docker-image) which contains all those dependencies.
TO
Instead of setting up the application dependencies in each environment (tools, libraries, configuration) we can use a [docker image](#docker-image) which contains all those dependencies.

FROM
This image is then consistent between development machine, cloud host, VPS, etc.
TO
This image is then consistent between development machine, cloud host and Virtual Private Servers.

FROM
[Docker container](#docker-container) is self-contained and isolated - other applications or different versions of some tool running on host won't affect container execution. 
TO
[Docker container](#docker-container) is self-contained and isolated which means other applications or different versions of some tool running on host won't affect container execution. 

FROM
Likewise once something is added to an image, even if the remote source is gone, the image contains all the files it needs to run.
TO
Once something is added to the image, the image contains all the files it needs to run, even if the remote source is gone,